### ISSUE TITLE
[RDS] manage tables and databases in CDK

### ISSUE THREAD

**Issue by jonny-rimek** [aws/aws-cdk#9439]
v1: specify a SQL file that describes the state of your databases etc, this would be similar to ruby on rails structure.sql file. Then you point CDK to the file, like fromAsset for ECS or lambda.

v2. manage the structure of your database directly in CDK, like DynamoDB tables are created in CDK

### Use Case

The state of the RDBMS is part of the infrastructure as much as the size of the instance itself. 
It feels more natural to handle this in IaC than in your application code and run something like `rake db:migrate` on every deployment. 

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by jonny-rimek** [comment]
I kinda forgot that you don't define the desired state with relational databases and the tables update themself to match that state, but you have to write SQL to `alter` the tables and as a result, get a new state.

This is pretty much the opposite of how IaC works, so it probably makes less sense to combine it, especially v2 makes less sense now.

Maybe there is still a place for a custom resource that executes migrations during development.

**Comment by skinny85** [comment]
Hey @jonny-rimek ,

thanks for opening the issue. I would say that you should use dedicated tools for this functionality, like [Flyway DB](https://flywaydb.org), or [Liquibase](https://www.liquibase.org). I don't think the CDK should be in business of adding those features, as it's a huge project in and of itself, and it's not really a core competency of the CDK as we understand it.

Let me know if this makes sense!

Thanks,
Adam

**Comment by jonny-rimek** [comment]
Hi @skinny85 ,

I agree it would just add bloat and take resources away from other places, also there are tons of good existing solutions to this problem.

Thanks for your feedback

**Comment by hoegertn** [comment]
While I totally agree with schema design and the use of Liquibase, I still think that databases and users might be a good fit for CDK because I would need these resources in the following constructs as env variables for Lambda, Fargate, etc.

**Comment by jonny-rimek** [comment]
Good point, I agree. How do you handle that right now, just update the information in two places?
